fsspec/implementations/http.py

```diff
@@ -316,22 +360,33 @@ async def _info(self, url, **kwargs):
         which case size will be given as None (and certain operations on the
         corresponding file will not work).
         """
-        size = False
-        kw = self.kwargs.copy()
-        kw.update(kwargs)
+        info = {}
+        failed = exc = None
+        session = await self.set_session()
+
         for policy in ["head", "get"]:
             try:
-                session = await self.set_session()
-                size = await _file_size(url, size_policy=policy, session=session, **kw)
-                if size:
+                info.update(
+                    await _file_info(
+                        url,
+                        size_policy=policy,
+                        session=session,
+                        **self.kwargs,
+                        **kwargs,
+                    )
+                )
+                if info.get("size") is not None:
                     break
             except Exception as e:
+                failed = policy
+                exc = e
```

Comment:
Future reference: any place where we capture the exception and carry it out of the block like this, we are likely making circular references via the traceback frames. This might cause memory leaks.